engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java

```diff
@@ -3516,6 +3517,16 @@ private void checkPlatformCompatible(AppPackage ap)
     }
   }
 
+  private void checkJavaVersionCompatible(AppPackage ap)
+  {
+    String buildJdkVersion = ap.getBuildJdkVersion();
+    String systemVersion = System.getProperty("java.version");
+    if (VersionInfo.compare(buildJdkVersion, systemVersion) > 0) {
```

Comment:
@oliverwnk The check for Build-jdk is not good. As @sanjaypujare pointed out, it is possible to build an application package on a machine where JDK 1.8 is installed and generate 1.7 or 1.6 compatible binaries. The check is way too restrictive.